src/fs/protocol/public/types.ts

```diff
@@ -11,9 +11,9 @@ export type PutDetails = AddResult & {
 }
 
 export type SkeletonInfo = {
-  cid: CID
-  userland: CID
-  metadata: CID
+  cid: CID | string
+  userland: CID | string
+  metadata: CID | string
```

Comment:
Why did you make these all `CID | string` instead of just `CID`?

Reply:
Because these'll be coming from DAG structures directly which can have a CID of either types.

Reply:
Hm. This forces us to handle `cid`, `userland` etc. being both string or CID at everywhere we're using these types.
I think it would be better if at the point where we're parsing something external we'd parse it into a canonical internal format, i.e. one with only `CID`s.
This *should* result in less usage of `decodeCID` overall, because it should only be used in the code that takes the "real-world" data. Everything internal wouldn't have to know about decodeCID anymore.

Reply:
Hmm I now remember why I did this exactly. If I don't add the `CID | string` to these types here, I have to loop through all the skeletons and switch out the CIDs. I attempted to do this, but that seemed to involve much more refactoring than just adding a few `decodeCID` calls (10 of them to be exact, I just counted). I'm just gonna leave it for now, since we're going to be refactoring this later anyway.

Reply:
I think the other good news is that the skeleton will be gone with wnfs2 anyway :sweat_smile: